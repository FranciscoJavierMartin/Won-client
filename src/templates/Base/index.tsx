import { Container } from '@/components/Container';
import Footer from '@/components/Footer';
import Menu from '@/components/Menu';
import * as S from './styles';

export type BaseProps = {
  children: React.ReactNode;
};

const Base: React.FC<BaseProps> = ({ children }) => (
  <S.Wrapper>
    <S.SectionMenu>
      <Menu />
    </S.SectionMenu>
    {children}
    <S.SectionFooter>
      <Container>
        <Footer />
      </Container>
    </S.SectionFooter>
  </S.Wrapper>
);

export default Base;
