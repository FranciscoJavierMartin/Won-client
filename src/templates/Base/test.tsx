import { renderWithTheme } from '@/utils/tests/helpers';
import { screen } from '@testing-library/react';
import Base from '.';

jest.mock('@/components/Menu', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Menu"></div>;
    },
  };
});

jest.mock('@/components/Footer', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Footer"></div>;
    },
  };
});

describe('<Base />', () => {
  it('should render menu, foote and children', () => {
    const { container } = renderWithTheme(
      <Base>
        <h1>Heading</h1>
      </Base>
    );

    expect(screen.getByTestId('Mock Menu')).toBeInTheDocument();
    expect(screen.getByTestId('Mock Footer')).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: /heading/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
