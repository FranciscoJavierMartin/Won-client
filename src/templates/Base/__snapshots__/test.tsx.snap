// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should render menu, foote and children 1`] = `
.c1 {
  width: 100%;
  max-width: 130rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(3.2rem / 2);
  padding-right: calc(3.2rem / 2);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  margin-top: 4.0rem;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c3 {
  margin-top: 4.0rem;
  padding-bottom: 1.6rem;
  padding-top: 5.6rem;
  background-color: #FAFAFA;
  -webkit-clip-path: polygon(0 5%,100% 0%,100% 100%,0 100%);
  clip-path: polygon(0 5%,100% 0%,100% 100%,0 100%);
}

@media (min-width:768px) {
  .c3 {
    padding-top: calc(5.6rem * 2);
    -webkit-clip-path: polygon(0 15%,100% 0%,100% 100%,0 100%);
    clip-path: polygon(0 15%,100% 0%,100% 100%,0 100%);
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      data-testid="Mock Menu"
    />
  </div>
  <div
    class="c2"
  >
    <h1>
      Heading
    </h1>
  </div>
  <section
    class="c3"
  >
    <div
      class="c1"
    >
      <div
        data-testid="Mock Footer"
      />
    </div>
  </section>
</div>
`;
